lib/logger.js

```diff
@@ -14,10 +14,11 @@ class Logger extends Transform {
     }
   }
 
-  writeLog (level, message) {
+  writeLog (level, message, details = {}) {
     this.push({
       stack: [this.node],
       level: level.toUpperCase(),
+      details: details,
```

Comment:
Maybe we define `name` as a predefined detail and use the property directly in this structure?